test/fake-intake/fakeintake/server.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fakeintake
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+type payload struct {
+	timestamp time.Time
+	data      []byte
+}
+
+type FakeIntake struct {
+	mu     sync.RWMutex
+	server http.Server
+
+	payloadStore map[string][]payload
+}
+
+// NewFakeIntake creates a new fake intake and starts it on localhost:5000
+func NewFakeIntake() *FakeIntake {
+	fi := &FakeIntake{
+		mu:           sync.RWMutex{},
+		payloadStore: map[string][]payload{},
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", fi.postPayload)
+	mux.HandleFunc("/fake/payloads/", fi.getPayloads)
+
+	fi.server = http.Server{
+		Addr:    ":5000",
+		Handler: mux,
+	}
+
+	fi.start()
+
+	return fi
+}
+
+func (fi *FakeIntake) start() {
+	go func() {
+		err := fi.server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			fmt.Printf("Error creating fake intake server at %s: %v", fi.server.Addr, err)
+		}
+	}()
+}
+
+// Stop Gracefully stop the http server
+func (fi *FakeIntake) Stop() error {
```

Comment:
Stop should take a context in parameter to ensure server is stopped in time or aborted (see also the comment about having a `Run(ctx)`.

Reply:
Will be address by AETT-83